code/datums/chatmessage.dm

```diff
@@ -1,59 +1,56 @@
-/// How long the chat message's spawn-in animation will occur for
-#define CHAT_MESSAGE_SPAWN_TIME 0.2 SECONDS
-/// How long the chat message will exist prior to any exponential decay
-#define CHAT_MESSAGE_LIFESPAN 5 SECONDS
-/// How long the chat message's end of life fading animation will occur for
-#define CHAT_MESSAGE_EOL_FADE 0.7 SECONDS
-/// Factor of how much the message index (number of messages) will account to exponential decay
-#define CHAT_MESSAGE_EXP_DECAY 0.7
-/// Factor of how much height will account to exponential decay
-#define CHAT_MESSAGE_HEIGHT_DECAY 0.9
-/// Approximate height in pixels of an 'average' line, used for height decay
-#define CHAT_MESSAGE_APPROX_LHEIGHT 11
-/// Max width of chat message in pixels
-#define CHAT_MESSAGE_WIDTH 96
-/// Max length of chat message in characters
-#define CHAT_MESSAGE_MAX_LENGTH 110
-/// The dimensions of the chat message icons
-#define CHAT_MESSAGE_ICON_SIZE 9
-
-///Base layer of chat elements
-#define CHAT_LAYER 1
-///Highest possible layer of chat elements
-#define CHAT_LAYER_MAX 2
-/// Maximum precision of float before rounding errors occur (in this context)
-#define CHAT_LAYER_Z_STEP 0.0001
-/// The number of z-layer 'slices' usable by the chat message layering
-#define CHAT_LAYER_MAX_Z (CHAT_LAYER_MAX - CHAT_LAYER) / CHAT_LAYER_Z_STEP
+///the layer value given to the message image in the last animation step after the image has faded.
+///used for either the initialization animation sequence or the edited stage 2 duration animation sequence.
+///to the server the images layer is set instantly to this value, but the client cant see it because its only adjusted for them
+/// after the message is inivisible. so this is used to mark what animation is being used for the server.
+#define MESSAGE_ANIMATION_DEFAULT_LAYER_MARK 1020
+///the layer value given to the message image for the animation sequence where the image is forced into fading.
+///used so that the server doesnt force the image to fade twice.
+#define MESSAGE_ANIMATION_FORCE_FADE_LAYER_MARK 1021
+///the layer mark given to the last stage of the animation sequence after it has been edited but not forced into the fading animation.
+#define MESSAGE_ANIMATION_EDIT_FADE_LAYER_MARK 1022
```

Comment:
no because its not real when you consider it affecting visuals that clients see. it never actually visually gets to that layer for the client because its only done after the message fade animations have already completed and the alpha becomes 0. to the server it does instantly set layer so we can use it to tell what animation we're in